streampipes-extensions/streampipes-connect-adapters-iiot/src/main/resources/org.apache.streampipes.connect.iiot.adapters.plc4x.s7/documentation.md

```diff
@@ -26,8 +26,62 @@
 
 ## Description
 
-Shows the live position of the International Space Station (ISS), updated every two seconds.
-
+The adapter allows to connect with a Siemens S7 PLC.
 
 ***
 
+## Configuration
+
+The following configuration options are available when creating an adapter:
+
+### PLC Address
+
+This field requires the PLC address in form of the IP without the prefixed protocol (e.g., 192.168.34.56).
+
+In addition to the pure IP, other parameters supported by Apache PLC4X can be provided as an URL parameter:
+
+* `local-rack`
+* `local-slot`
+* `local-tsap`
+* `remote-rack`
+* `remote-slot`
+
+Additional configs are separated by `&`.
+
+Example address: `192.68.34.56?remote-rack=0&remote-slot=3&controller-type=S7_400`
+
+See the <a href="https://plc4x.apache.org/users/protocols/s7.html">Apache PLC4X documentation</a> for more information.
+
+### Polling Interval
+
+The polling interval requires a number in milliseconds, which represents the interval in which the adapter will poll the
+PLC for new data. For instance, a polling interval of 1000 milliseconds will configure the adapter to send a request to
+the PLC every second.
+
+### Nodes
+
+In the Nodes section, the PLC nodes that should be gathered are defined.
+There are two options to define the nodes:
+
+* Manual configuration: The address must be assigned manually by providing a runtime name, the node name and the
+  datatype. The `Runtime Name` will be the StreamPipes-internal name of the field, which will also show up in the data
+  explorer and pipeline editor. The `Node Name` refers to the node address of the PLC, e.g., `%Q0.4`. Finally, the data
+  type can be selected from the available selection. Currently available data types
+  are `Bool`, `Byte`, `Int`, `Word`, `Real`, `Char`, `String`, `Date`, `Time of Day` and `Date and Time`.
+* Instead of providing the node information manually, a CSV file can be uploaded. The CSV file can, for instance, be
+  exported from TIA and then be enriched with the appropriate runtime names. This is especially useful when many fields
+  should be added as nodes. Here is an example export enriched with the runtime name:
+
+```
+Runtime Name,Path,Data Type,Node Name,Comment,Hmi Visible,Hmi Accessible,Hmi Writeable,Typeobject ID,Version ID
```

Comment:
Only the `Runtime Name`, `Data Type`, and `Node Name` columns are required. The mapping is done based on the title of the corresponding`StaticProperty`. All other fields are ignored. I'd suggest removing the other columns for simplicity. Maybe we can add a comment if there are more columns it is not a problem because they will be ignored.

Reply:
I left one additional field so that it is clear that the CSV  does not exactly need to contain the required fields.